includes/functions.inc.php

```diff
@@ -1670,21 +1670,30 @@ function tag_cloud($days, $scale_min, $scale_max) {
 /**
  * converts a unix timestamp into a formated date string
  *
- * @param string $format : like parameter for DateTime::format() - https://www.php.net/manual/de/datetime.format.php
- * @param int $timestamp : UNIX timestamp
+ * @param string $format : like parameter for IntlDateFormatter::format
+ * @param long $timestamp : UNIX timestamp
  * @return string
  */
 function format_time($format, $timestamp = 0) {
+	// https://www.php.net/manual/de/function.setlocale.php
+	// https://www.php.net/manual/de/intldateformatter.format.php
+	$formatter = new IntlDateFormatter(
+		setlocale(LC_TIME, '0'),
+		IntlDateFormatter::FULL,
+		IntlDateFormatter::FULL,
+	);
+	$formatter->setPattern($format);
+ 
 	if ($timestamp == 0) 
 		$timestamp = TIMESTAMP;
-	
-	if (defined('LOCALE_CHARSET')) {
-		return iconv(LOCALE_CHARSET,CHARSET, date($format,$timestamp));
-	}
-	else {
-		return date($format, $timestamp);
-	}
-}
+ 
+	$date = date_create();
+	$dateTime = date_timestamp_set($date, $timestamp);
+ 
+	if (defined('LOCALE_CHARSET'))
```

Comment:
Do we really need this optional step to convert the charset? I would assume, that the source and the target charset is UTF-8 in any case. Or do I miss something?

Reply:
If so, why one can specify the charset in lang files? In any case, this condition was already part of the function and is not part of my modification. ;-)
Let me know, if I should remove the condition.

Reply:
No, leave it as it is. This was more a general consideration. Sorry for bringing it into the review.

Reply:
Thank you!